### PR TITLE
doc: update changelogs for 2.1.0 release

### PR DESCRIPTION
**Pull request by greg-fer**
Updated changelogs in nrfxlib with 2.1.0 sections.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

**Comment by umapraseeda** [comment]
Could you update MPSL changelog also?

**Comment by greg-fer** [comment]
> Could you update MPSL changelog also?

@umapraseeda , done.

**Review by umapraseeda — APPROVED** [review]

**Review by grochu — APPROVED** [review]
Looks good

File: nrf_802154/doc/rd_limitations.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+In addition, see :ref:`known_issues` for temporary issues that will be fixed in future releases.
+
+IRQ priority limitations
+  Application and device drivers (excluding those compliant with :ref:`mpsl`) must not use IRQ priority higher than :c:macro:`NRF_802154_SWI_PRIORITY` and :c:macro:`NRF_802154_SL_RTC_IRQ_PRIORITY`.
+
+KRKNWK-11204: Transmitting an 802.15.4 frame with improperly populated Auxiliary Security Header field will result in assert
+  **Workaround:** Make sure that you populate the Auxiliary Security Header field according to the IEEE Std 802.15.4-2015 specification, section 9.4.
+
+KRKNWK-12482: Reception of correct frames will occasionally end in failure with error ``NRF_802154_RX_ERROR_RUNTIME``
+  This issue can occur for the ``nrf5340dk_nrf5340_cpunet`` target if a custom application (other than :ref:`multiprotocol-rpmsg-sample` sample or `nRF IEEE 802.15.4: Serialization RPMsg <https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/samples/boards/nrf/ieee802154/802154_rpmsg/README.html>`_ sample) is used.
```
Comment (b-gent) [inline nrf_802154/doc/rd_limitations.rst:16]: We have a links.txt file in nrfxlib as well, please define the link there

**Review by b-gent — CHANGES_REQUESTED** [review]

File: nrf_802154/doc/rd_limitations.rst (hunk 1 above)
Comment (greg-fer) [reply nrf_802154/doc/rd_limitations.rst]: We're updating dependencies in CMakeLists and conf.py here instead to enable linking with `zephyr:` -- https://github.com/nrfconnect/sdk-nrf/pull/8510

**Review by b-gent — APPROVED** [review]

**Comment by umapraseeda** [comment]
@greg-fer There is a 'main branch' entry in lc3..might need to fix that.

**Review by rugeGerritsen — APPROVED** [review]

**Pull request merged**